client/components/search/index.jsx

```diff
@@ -271,19 +280,26 @@ const Search = React.createClass( {
 			spellCheck: 'false'
 		};
 
-		searchClass = classNames( this.props.additionalClasses, {
+		const searchClass = classNames( this.props.additionalClasses, this.props.dir, {
 			'is-expanded-to-container': this.props.fitsContainer,
 			'is-open': isOpenUnpinnedOrQueried,
 			'is-searching': this.props.searching,
-			rtl: this.props.dir === 'rtl',
-			ltr: this.props.dir === 'ltr',
+			'no-close-button': this.props.hideClose,
```

Comment:
This is a new thing. We want in some scenarios to not show close at all. Take a look at https://teamhyperion.wordpress.com/2016/08/15/calypso-theme-showcase-magic-search-design-i5/